runtime/adapters/src/price.rs

```diff
@@ -55,28 +56,83 @@ where
 		// if this gets removed (eg. Convert returns weight), the constraint on T and ema-oracle is not necessary
 		price_weight.saturating_accrue(pallet_ema_oracle::Pallet::<T>::get_price_weight().saturating_mul(2));
 
+		//2 reads as we are checking if from and to assets are transaction fee currencies
+		let is_transaction_fee_currency_check_weight = T::DbWeight::get().reads(2);
+		price_weight.saturating_accrue(is_transaction_fee_currency_check_weight);
+
 		let Some((converted, _)) = C::convert((from_currency, to_currency, account_balance)) else {
 			return (0, price_weight);
 		};
 		(converted, price_weight)
 	}
 }
 
-pub struct ConvertAmount<P>(sp_std::marker::PhantomData<P>);
+pub struct ConvertAmount<PriceProv, SwappablePaymentAssetSupport, DotAssetId>(
+	sp_std::marker::PhantomData<(PriceProv, SwappablePaymentAssetSupport, DotAssetId)>,
+);
```

Comment:
nitpick: this one is one me - it should be really be called "ConvertBalance" ( considering my other nitpick here)